lib/solveEquation/stepThrough.js

```diff
@@ -182,12 +297,35 @@ function step(equation, symbolName) {
 function addSimplificationSteps(steps, equation, debug=false) {
   let oldEquation = equation.clone();
 
+  /*
+    Collects substeps for factoring and simplifying the expression separately
+    because their logics are located in different places. Collect factoring
+    substeps first in case the original expression is factorable. If no factoring
+    substeps, proceed to collect simplifying substeps.
+    Some examples of when this comes into play:
+    e.g. x * x + 3x + 2 = 0 <- factor the left side
+    e.g. x + 4 + 2 = 0 <- simplify the left side
+  */
   const leftSteps = simplifyExpressionNode(equation.leftNode, false);
```

Comment:
For clarity, can you call this `leftSimplifySteps` (and same for the right hand part)